sdk/core/System.ClientModel/src/Message/MultipartContent.cs

```diff
@@ -0,0 +1,295 @@
+    /// </summary>
+    public string Boundary => _boundary;
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="MultipartContent"/> class with a randomly generated boundary.
+    /// </summary>
+    public MultipartContent() : this(Guid.NewGuid().ToString())
+    {
+    }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="MultipartContent"/> class with the specified boundary.
+    /// </summary>
+    /// <param name="boundary">The boundary string for the multipart content, which is used to separate the body parts.</param>
+    public MultipartContent(string boundary)
+    {
+        ValidateBoundary(boundary);
+
+        _boundary = boundary;
+    }
+
+    /// <summary>
+    /// Gets the <see cref="ContentType"/> of this multipart body, to be used when formatting the request.
+    /// </summary>
+    /// <param name="contentType"></param>
+    public ContentType GetContentType(ContentType contentType)
+    {
+        if (contentType == null)
+        {
+            throw new ArgumentNullException(nameof(contentType));
+        }
+
+        contentType.Boundary = _boundary;
+        return contentType;
+    }
+
+    /// <summary>
+    /// Returns the content of the <see cref="MultipartContent"/> as a <see cref="BinaryContent"/>.
+    /// </summary>
+    public BinaryContent ToBinaryContent() => new MultipartBinaryContent(_subparts, _boundary);
+
+    /// <summary>
+    /// Adds a new <see cref="BinaryContent"/> instance to the collection of <see cref="BinaryContent"/> objects that get
+    /// formatted into multipart/* content.
+    /// </summary>
+    /// <param name="content">The <see cref="BinaryContent"/> to add to the collection.</param>
+    public void Add(BinaryContent content) => Add(content, new Dictionary<string, string>());
+
+    /// <summary>
+    /// Adds a new <see cref="BinaryContent"/> instance to the collection of <see cref="BinaryContent"/> objects that get
+    /// formatted into multipart/* content.
+    /// </summary>
+    /// <param name="content">The <see cref="BinaryContent"/> to add to the collection.</param>
+    /// <param name="headers">The headers to add to this <see cref="BinaryContent"/>'s section.</param>
+    public void Add(BinaryContent content, Dictionary<string, string> headers)
+    {
+        if (content == null)
+        {
+            throw new ArgumentNullException(nameof(content));
+        }
+
+        if (headers == null)
+        {
+            throw new ArgumentNullException(nameof(headers));
+        }
+
+        var subpart = new MultipartContentSubpart(content, headers);
+
+        _subparts.Add(new MultipartContentSubpart(content, headers));
+    }
+
+    private static void ValidateBoundary(string boundary)
+    {
+        if (string.IsNullOrWhiteSpace(boundary))
+        {
+            throw new ArgumentException("Value cannot be null or empty.", boundary);
```

Comment:
Should this use `nameof`?

```suggestion
            throw new ArgumentException("Value cannot be null or empty.", nameof(boundary));
```